modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/LongGaugeAdapter.java

```diff
@@ -10,37 +10,46 @@
 
 import io.opentelemetry.api.metrics.Meter;
 
+import java.util.Collections;
 import java.util.Map;
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * LongGaugeAdapter wraps an otel ObservableLongMeasurement
  */
-public class LongGaugeAdapter extends AbstractInstrument<io.opentelemetry.api.metrics.ObservableLongMeasurement>
+public class LongGaugeAdapter extends AbstractInstrument<io.opentelemetry.api.metrics.ObservableLongGauge>
     implements
         org.elasticsearch.telemetry.metric.LongGauge {
+    private final AtomicReference<ValueWithAttributes<Long>> valueWithAttributes;
```

Comment:
Why not just do a new record here?  That will avoid boxing as well. 
`record LongValueWithAttribute(long value, Map<String, Object> attributes)`.